staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go

```diff
@@ -150,6 +151,59 @@ var (
 		[]string{"requestKind"},
 	)
 
+	// low and high concurrency watermarks from the last few windows
+	concurrencyWatermarks = compbasemetrics.NewGaugeVec(
+		&compbasemetrics.GaugeOpts{
+			Name:           "apiserver_inflight_count_watermarks",
+			Help:           "Low and high watermark of mutating, readonly requests executing in recent windows.",
+			StabilityLevel: compbasemetrics.ALPHA,
+		},
+		[]string{"requestKind", "mark", "lag"},
+	)
+
+	concurrencyAverage = compbasemetrics.NewGaugeVec(
+		&compbasemetrics.GaugeOpts{
+			Name:           "apiserver_inflight_count_average",
+			Help:           "Average count of mutating, readonly requests executing during recent windows.",
+			StabilityLevel: compbasemetrics.ALPHA,
+		},
+		[]string{"requestKind"},
+	)
+
+	concurrencyStddev = compbasemetrics.NewGaugeVec(
+		&compbasemetrics.GaugeOpts{
+			Name:           "apiserver_inflight_count_stddev",
+			Help:           "Standard deviation of count of mutating, readonly requests executing during recent windows.",
+			StabilityLevel: compbasemetrics.ALPHA,
+		},
+		[]string{"requestKind"},
+	)
+
+	inflightCountElapsedSecondses = compbasemetrics.NewCounterVec(
+		&compbasemetrics.CounterOpts{
+			Name:           "apiserver_inflight_count_seconds",
+			Help:           "Seconds over which count of mutating, readonly requests executing have been integrated.",
+			StabilityLevel: compbasemetrics.ALPHA,
+		},
+		[]string{"requestKind"},
+	)
+	inflightCountIntegrals = compbasemetrics.NewCounterVec(
+		&compbasemetrics.CounterOpts{
+			Name:           "apiserver_inflight_count_integral",
+			Help:           "Integral of count of mutating, readonly requests executing since startup.",
+			StabilityLevel: compbasemetrics.ALPHA,
+		},
+		[]string{"requestKind"},
+	)
+	inflightCountSquaredIntegrals = compbasemetrics.NewCounterVec(
+		&compbasemetrics.CounterOpts{
+			Name:           "apiserver_inflight_count_squared_integral",
```

Comment:
this should be entirely calculated in PromQL queries based on counts

Reply:
What would that PromQL expression be?